sfge/src/main/java/com/salesforce/graph/ops/ApexValueUtil.java

```diff
@@ -51,28 +52,57 @@ public static Optional<ApexValue<?>> applyBinaryExpression(
         } else {
             rhs = ApexValueBuilder.get(symbols).valueVertex(vertex.getRhs()).build();
         }
+        checkNullAccess(rhs, vertex);
+
+        if (vertex.getOperator().equalsIgnoreCase(ASTConstants.OPERATOR_ADDITION)
+                && lhs instanceof ApexStringValue
+                && rhs instanceof ApexStringValue) {
+            return performStringConcatenation(
+                    (ApexStringValue) lhs, (ApexStringValue) rhs, vertex, symbols);
+        }
+
+        // TODO: Handle more binary expression scenarios
+
+        if (LOGGER.isDebugEnabled()) {
+            // This can be because either the binary expression has not been resolved or the are
+            // non-strings
+            LOGGER.debug("Unable to apply binary expression. vertex=" + vertex);
+        }
+        return Optional.empty();
+    }
 
-        if (lhs instanceof ApexStringValue
-                && lhs.isValuePresent()
-                && rhs instanceof ApexStringValue
-                && rhs.isValuePresent()) {
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("Applying: lhs=" + lhs + ", rhs=" + rhs);
+    private static void checkNullAccess(ApexValue<?> apexValue, BinaryExpressionVertex vertex) {
+        // String value get represented by "null" when used in a concatenation scenario.
+        //  Therefore, we exclude String values from null access checks.
+        if (apexValue != null && !(apexValue instanceof ApexStringValue)) {
+            if (apexValue.isNull()) {
+                throw new NullValueAccessedException(apexValue, vertex);
             }
+        }
+    }
+
+    private static Optional<ApexValue<?>> performStringConcatenation(
+            ApexStringValue lhs,
+            ApexStringValue rhs,
+            BinaryExpressionVertex vertex,
+            SymbolProvider symbols) {
+        final String lhsValue =
+                lhs.isValuePresent() ? lhs.getValue().get() : lhs.isNull() ? "null" : UNKNOWN;
+        final String rhsValue =
+                rhs.isValuePresent() ? rhs.getValue().get() : rhs.isNull() ? "null" : UNKNOWN;
```

Comment:
we were previously ignoring this null altogether. Apex, however, treats null strings as "null". I've made relevant changes here.